core/src/main/java/io/questdb/griffin/engine/functions/table/TouchTableFunctionFactory.java

```diff
@@ -0,0 +1,171 @@
+            arg.init(symbolTableSource, executionContext);
+            this.sqlExecutionContext = executionContext;
+        }
+
+        @Override
+        public CharSequence getStr(Record rec) {
+            sinkA.clear();
+            getStr(rec, sinkA);
+            return sinkA;
+        }
+
+        @Override
+        public CharSequence getStrB(Record rec) {
+            sinkB.clear();
+            getStr(rec, sinkB);
+            return sinkB;
+        }
+
+        @Override
+        public void getStr(Record rec, CharSink sink) {
+            touchTable();
+            sink.put("{\"data_pages\": ")
+                    .put(dataPages)
+                    .put(", \"index_key_pages\":")
+                    .put(indexKeyPages)
+                    .put( ", \"index_values_pages\": ")
+                    .put(indexValuePages).put("}");
+        }
+
+        private void clearCounters() {
+            garbage = 0;
+            dataPages = 0;
+            indexKeyPages = 0;
+            indexValuePages = 0;
+        }
+
+        private void touchTable() {
+            clearCounters();
+            final long pageSize = Files.PAGE_SIZE;
+            try (RecordCursorFactory recordCursorFactory = arg.getRecordCursorFactory()) {
+                try (PageFrameCursor pageFrameCursor = recordCursorFactory.getPageFrameCursor(sqlExecutionContext)) {
+                    PageFrame frame;
+                    RecordMetadata metadata = recordCursorFactory.getMetadata();
+                    while ((frame = pageFrameCursor.next()) != null) {
+                        for (int columnIndex = 0, sz = metadata.getColumnCount(); columnIndex < sz; columnIndex++) {
+
+                            final long columnMemorySize = frame.getPageSize(columnIndex);
+                            final long columnBaseAddress = frame.getPageAddress(columnIndex);
+                            dataPages += touchMemory(pageSize, columnBaseAddress, columnMemorySize);
+
+                            if (metadata.isColumnIndexed(columnIndex)) {
+                                final BitmapIndexReader indexReader = frame.getBitmapIndexReader(columnIndex, BitmapIndexReader.DIR_BACKWARD);
+
+                                final long keyBaseAddress = indexReader.getKeyBaseAddress();
+                                final long keyMemorySize = indexReader.getKeyMemorySize();
+                                indexKeyPages += touchMemory(pageSize, keyBaseAddress, keyMemorySize);
+
+                                final long valueBaseAddress = indexReader.getValueBaseAddress();
+                                final long valueMemorySize = indexReader.getValueMemorySize();
+                                indexValuePages += touchMemory(pageSize, valueBaseAddress, valueMemorySize);
+                            }
+                        }
+                    }
+                } catch (SqlException ignored) {
+                    // do not propagate
```

Comment:
I would propagate this or at very least log it

Reply:
I've added logging as exception propagation requires signature change in Function interface